src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -0,0 +1,354 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Business;
+using Bit.Core.OrganizationFeatures.OrganizationSubscriptionUpdate.Interface;
+using Bit.Core.Repositories;
+using Bit.Core.SecretsManager.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Tools.Enums;
+using Bit.Core.Tools.Models.Business;
+using Bit.Core.Tools.Services;
+using Bit.Core.Utilities;
+using Microsoft.Extensions.Logging;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationSubscriptionUpdate;
+
+public class UpdateSecretsManagerSubscriptionCommand : IUpdateSecretsManagerSubscriptionCommand
+{
+    private readonly IOrganizationRepository _organizationRepository;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+    private readonly IPaymentService _paymentService;
+    private readonly IOrganizationService _organizationService;
+    private readonly ICurrentContext _currentContext;
+    private readonly IMailService _mailService;
+    private readonly ILogger<UpdateSecretsManagerSubscriptionCommand> _logger;
+    private readonly IServiceAccountRepository _serviceAccountRepository;
+    private readonly IReferenceEventService _referenceEventService;
+
+    public UpdateSecretsManagerSubscriptionCommand(
+        IOrganizationRepository organizationRepository,
+        IOrganizationService organizationService,
+        IOrganizationUserRepository organizationUserRepository,
+        IPaymentService paymentService,
+        ICurrentContext currentContext,
+        IMailService mailService,
+        ILogger<UpdateSecretsManagerSubscriptionCommand> logger,
+        IServiceAccountRepository serviceAccountRepository,
+        IReferenceEventService referenceEventService)
+    {
+        _organizationRepository = organizationRepository;
+        _organizationUserRepository = organizationUserRepository;
+        _paymentService = paymentService;
+        _organizationService = organizationService;
+        _currentContext = currentContext;
+        _mailService = mailService;
+        _logger = logger;
+        _serviceAccountRepository = serviceAccountRepository;
+        _referenceEventService = referenceEventService;
+    }
+
+    public async Task UpdateSecretsManagerSubscription(OrganizationUpdate update)
+    {
+        var organization = await _organizationRepository.GetByIdAsync(update.OrganizationId);
+
+        if (organization == null)
+        {
+            throw new NotFoundException();
+        }
+
+        var newSeatCount = organization.SmSeats.HasValue
+            ? organization.SmSeats.Value + update.SeatAdjustment
+            : update.SeatAdjustment;
```

Comment:
This would be a more idiomatic way of writing this:

```suggestion
        var newSeatCount = organization.SmSeats.GetValueOrDefault() + update.SeatAdjustment;
```

Reply:
thanks